src/function/probability/lgamma.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-loss-of-precision */
+
+import { lgammaNumber, lnSqrt2PI } from '../../plain/number/index.js'
+import { deepMap } from '../../utils/collection.js'
+import { factory } from '../../utils/factory.js'
+
+const name = 'lgamma'
+const dependencies = ['Complex', 'typed']
+
+export const createLgamma = /* #__PURE__ */ factory(name, dependencies, ({ Complex, typed }) => {
+  const SMALLX = 7
+  const SMALLY = 7
+
+  /**
+   * Logarithm of the gamma function for real, positive numbers.
+   * Based off of Tom Minka's lightspeed toolbox.
+   *
+   * Syntax:
+   *
+   *    math.lgamma(n)
+   *
+   * Examples:
+   *
+   *    math.lgamma(5)       // returns 3.178053830347945
+   *    math.lgamma(0)       // returns Infinity
+   *    math.lgamma(-0.5)    // returns NaN
+   *
+   * See also:
+   *
+   *    gamma
+   *
+   * @param {number | Array | Matrix} n   A real or complex number
+   * @return {number | Array | Matrix}    The log gamma of `n`
+   */
+
+  return typed(name, {
+    number: lgammaNumber,
+
+    Complex: function (n) {
+      if (n.isNaN()) {
+        return new Complex(NaN, NaN)
+      }
+
+      if (n.im === 0) {
+        return this(n.re)
+      }
```

Comment:
Actually I don't think we should return a real number here, even now I just follow the way that `gamma` uses, [gamma.js#L39](https://github.com/josdejong/mathjs/blob/develop/src/function/probability/gamma.js#L39).

If we do this, users will have to check if the returned value is real or complex, the code `mathjs.lgamma(x).doSomethingElse()` will fail infrequently.

I think we can just
```js
if (n.im === 0) {
  return new Complex(this(n.re), 0)
}
```

Any ideas? may I change the code?

Reply:
That makes sense, good point. 

In general, in mathjs there are places where for example numeric input like `sqrt(-4)` are "upgraded" into a Complex number, but not yet the other way around by "downgrading" to a regular number again. This can be configured with the `predictable` config. It makes sense to me though to keep the behavior in this case predictable by just always giving complex output when the input is complex.